code/modules/mob/living/silicon/robot/robot_models.dm

```diff
@@ -139,7 +198,31 @@
 	qdel(src)
 	return RM
 
-/obj/item/robot_model/proc/be_transformed_to(obj/item/robot_model/old_model)
+/obj/item/robot_model/proc/be_transformed_to(obj/item/robot_model/old_model, forced = FALSE)
+	if(islist(borg_skins) && !forced)
+		var/mob/living/silicon/robot/cyborg = loc
+		var/list/reskin_icons = list()
+		for(var/skin in borg_skins)
+			var/list/details = borg_skins[skin]
+			reskin_icons[skin] = image(icon = details[SKIN_ICON], icon_state = details[SKIN_ICON_STATE])
+		var/borg_skin = show_radial_menu(cyborg, cyborg, reskin_icons, custom_check = CALLBACK(src, .proc/check_menu, cyborg, old_model), radius = 38, require_near = TRUE)
+		if(!borg_skin)
+			return FALSE
+		var/list/details = borg_skins[borg_skin]
+		if(!isnull(details[SKIN_ICON_STATE]))
```

Comment:
just some food for thought you could save yourself a zillion trillion null checks by having defaults on the `details` list, overridden by the borg skin? not sure if its worth bothering

Reply:
some borg models override stuff by themselves and thatwould fuck with this